src/PaddedViews.jl

```diff
@@ -168,6 +168,18 @@ Base.parent(A::PaddedView) = A.data
     return A.fillvalue
 end
 
+# Checking if a linear index is in the valid region is not efficient, so we only support
+# IndexCartesian method here
+function Base.setindex!(A::PaddedView{T, N}, v, i::Vararg{Int, N}) where {T, N}
+    @boundscheck checkbounds(A, i...)
+    setindex!(A.data, v, i...)
+end
+function Base.setindex!(A::PaddedView{T, 1}, v, i::Int) where {T}
+    @boundscheck checkbounds(A, i...)
+    setindex!(A.data, v, i...)
+end
+Base.setindex!(A::PaddedView{T, N}, v, i::Int) where {T, N} = throw(ArgumentError("`setindex!` with IndexLinear `$i` is not supported for `PaddedView`. You may convert it to `CartesianIndex` first."))
```

Comment:
```suggestion
Base.setindex!(A::PaddedView{T, N}, v, i::Int) where {T, N} = throw(ArgumentError("for performance reasons, linear indexing is not supported for multidimensional `PaddedView`s. Try using a cartesian indexing scheme instead."))
```
My reasoning is that `setindex!` should appear in the stacktrace. It would be good to harmonize this with the message for `getindex`, but I didn't look up what that is.